client/components/domains/trademark-claims-notice/index.jsx

```diff
@@ -0,0 +1,563 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Fragment } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { defer, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import { composeAnalytics, recordGoogleEvent, recordTracksEvent } from 'state/analytics/actions';
+import HeaderCake from 'components/header-cake';
+import Button from 'components/button';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class TrademarkClaimsNotice extends React.Component {
+	static propTypes = {
+		isSignupStep: PropTypes.bool,
+		onAccept: PropTypes.func,
+		onReject: PropTypes.func,
+		recordAcknowledgeTrademarkButtonClickInTrademarkNotice: PropTypes.func,
+		recordChooseAnotherDomainButtonClickInTrademarkNotice: PropTypes.func,
+		recordShowTrademarkNoticeButtonClickInTrademarkNotice: PropTypes.func,
+		suggestion: PropTypes.object,
+		trademarkClaimsNoticeInfo: PropTypes.object,
+	};
+
+	static defaultProps = {};
+
+	state = this.getDefaultState();
+
+	getDefaultState() {
+		return {
+			hasScrolledToBottom: false,
+			showFullNotice: false,
+		};
+	}
+
+	enableActionButtons = () => {
+		this.setState( { hasScrolledToBottom: true } );
+		window.removeEventListener( 'scroll', this.handleScroll );
```

Comment:
You need to call this method as well when this component is unmounted. Otherwise, it's possible to get a confusing `Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.` warning.

For example, select `dairyqueen.dev`, get the TCN view, click to show the notice, but _don't_ scroll down to the bottom. Navigate away (for example, by clicking the Back or Domains buttons) and scroll down to the page. The warning will be triggered, because the `handleScroll` callback was triggered, but the component is not mounted anymore.